libvast/src/system/node.cpp

```diff
@@ -27,192 +29,207 @@
-    rp.deliver(make_error(ec::syntax_error, "no endpoint given"));
-    return;
-  }
-  auto ep = to<endpoint>(args.get_as<std::string>(0));
-  if (!ep) {
-    rp.deliver(make_error(ec::parse_error, "invalid endpoint format"));
-    return;
-  }
+// Tries to establish peering to another node.
+caf::message peer_command(const command&, caf::actor_system& sys,
+                          caf::config_value_map& options,
+                          command::argument_iterator first,
+                          command::argument_iterator last) {
+  VAST_ASSERT(this_node != nullptr);
+  if (std::distance(first, last) != 1)
+    return err(ec::syntax_error, "expected exactly one endpoint argument");
+  auto ep = to<endpoint>(*first);
+  if (!ep)
+    return err(ec::parse_error, "invalid endpoint format");
   // Use localhost:42000 by default.
   if (ep->host.empty())
     ep->host = "127.0.0.1";
   if (ep->port == 0)
     ep->port = 42000;
-  VAST_DEBUG(self, "connects to", ep->host << ':' << ep->port);
-  auto& mm = self->system().middleman();
+  VAST_DEBUG(this_node, "connects to", ep->host << ':' << ep->port);
+  auto& mm = sys.middleman();
+  // TODO: this blocks the node, consider talking to the MM actor instead.
   auto peer = mm.remote_actor(ep->host.c_str(), ep->port);
   if (!peer) {
-    VAST_ERROR(self, "failed to connect to peer:",
-               self->system().render(peer.error()));
-    rp.deliver(peer.error());
-    return;
+    VAST_ERROR(this_node, "failed to connect to peer:",
+               sys.render(peer.error()));
+    return caf::make_message(std::move(peer.error()));
   }
-  VAST_DEBUG(self, "sends peering request");
-  auto t = actor_cast<actor>(self->state.tracker);
-  rp.delegate(*peer, peer_atom::value, t, self->state.name);
+  VAST_DEBUG(this_node, "sends peering request");
+  auto node_name = get_or(options, "node.name", "node");
+  auto& st = this_node->state;
+  this_node->delegate(*peer, peer_atom::value, st.tracker, st.name);
+  return caf::none;
 }
 
-void status(node_ptr self, message /* args */) {
-  auto rp = self->make_response_promise();
-  self->request(self->state.tracker, infinite, get_atom::value).then(
-    [=](const registry& reg) mutable {
-      json::object result;
+// Queries registered components for various status information.
+caf::message status_command(const command&, caf::actor_system&,
+                            caf::config_value_map&, command::argument_iterator,
+                            command::argument_iterator) {
+  auto rp = this_node->make_response_promise();
+  this_node->request(this_node->state.tracker, infinite, get_atom::value).then(
+    [=, self = this_node](const registry& reg) mutable {
+      json::object obj;
       for (auto& peer : reg.components) {
         json::array xs;
         for (auto& pair : peer.second)
           xs.push_back(json{pair.second.label});
-        result.emplace(peer.first, std::move(xs));
+        obj.emplace(peer.first, std::move(xs));
```

Comment:
I actually find `result` more meaningful, as it indicates right away that the value is being returned later on. Why did you prefer `obj`?